lib/AST/Decl.cpp

```diff
@@ -1433,6 +1433,23 @@ AbstractStorageDecl::getAccessStrategy(AccessSemantics semantics,
   llvm_unreachable("bad access semantics");
 }
 
+bool ValueDecl::isInPrivateOrLocalContext() const {
+  const DeclContext *DC = getDeclContext();
+  if (!DC->isTypeContext()) {
+    assert((DC->isModuleScopeContext() || DC->isLocalContext()) &&
+           "unexpected context kind");
+    return DC->isLocalContext();
+  }
+
+  auto *nominal = DC->getAsNominalTypeOrNominalTypeExtensionContext();
+  if (nominal == nullptr)
+    return false;
+
+  if (nominal->isOutermostPrivateOrFilePrivateScope())
```

Comment:
This is redundantly going to check all of *this* nominal's parents, which isn't necessary. Better to just put the "<= FilePrivate" check in verbatim.